libcaf_net/caf/net/web_socket/framing.hpp

```diff
@@ -152,10 +152,12 @@ class CAF_NET_EXPORT framing : public octet_stream::upper_layer,
     shutdown(err);
   }
 
+  bool validate_fragmented_utf8() noexcept;
```

Comment:
From the function name, I would expect this to take a "fragmented UTF-8" as input and then validate it. But this function does bookkeeping, so maybe we find a name that makes this more clear? Also should have some documentation on it, since it's doing non-obvious side effects.

```suggestion
  /// Checks whether the current input is valid UTF-8. Stores the last position
  /// while scanning in order to avoid validating the same bytes again.
  bool payload_valid() noexcept;
```